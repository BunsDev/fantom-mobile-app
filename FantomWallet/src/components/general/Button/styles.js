import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  buttonStyle: {
    backgroundColor: 'rgb(63,108,175)',
    // padding: 10,
    height: 60,
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },
  textStyle: {
    color: '#fff',
    fontSize: 20,
    // fontFamily: 'Times New Roman',
    fontWeight: 'bold',
  },
});