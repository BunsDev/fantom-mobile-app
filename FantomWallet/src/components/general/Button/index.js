// @flow
import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import styles from './styles';

type Props = {
  activeOpacity: number,
  text: string,
  onPress: () => void,
  buttonStyle: { [key: string]: string },
  textStyle: { [key: string]: string },
};
/**
 * Button : This is generic component , meant for rendering Button on any screen.
 */
const Button = ({
  activeOpacity,
  text,
  onPress = () => {},
  buttonStyle = {},
  textStyle = {},
}: Props) => (
  <TouchableOpacity
    style={{ ...styles.buttonStyle, ...buttonStyle }}
    activeOpacity={activeOpacity}
    onPress={onPress}
  >
    <Text style={{ ...styles.textStyle, ...textStyle }}>{text}</Text>
  </TouchableOpacity>
);

export default Button;
