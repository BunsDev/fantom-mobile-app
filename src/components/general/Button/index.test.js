import React from 'react';
import { shallow } from 'enzyme';

import Button from './index';

test('renders correctly', () => {
  const wrapper = shallow(
    <Button
      activeOpacity={1}
      text="Click here"
      onPress={() => { }}
      buttonStyle={{ backgroungColor: 'red' }}
      textStyle={{ padding: 2 }}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});
