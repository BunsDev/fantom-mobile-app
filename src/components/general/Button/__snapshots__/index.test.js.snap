// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={1}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(63,108,175)",
      "backgroungColor": "red",
      "height": 60,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 20,
        "fontWeight": "bold",
        "padding": 2,
      }
    }
  >
    Click here
  </Text>
</TouchableOpacity>
`;
